lib/ansible/modules/cloud/amazon/route53_zone.py

```diff
@@ -204,11 +204,12 @@ def create_private(conn, matching_zones, vpc_id, vpc_region, zone_in, record):
             changed = False
             return changed, record
 
-    result = conn.create_hosted_zone(zone_in, **record)
-    hosted_zone = result['CreateHostedZoneResponse']['HostedZone']
-    zone_id = hosted_zone['Id'].replace('/hostedzone/', '')
-    record['zone_id'] = zone_id
-    record['name'] = zone_in
+    if not module.check_mode:
```

Comment:
Thanks @ryansb. Is there an 'Ansible module best practices' or similar that has guidance on the preferred way to structure a module that supports check mode? My python-fu and time are both relatively limited - this PR was following some basic instructions off of a Google Groups post from Mr DeHaan in 2013 that I found during a search.

As a subject for discussion - all modules should support check mode from the first commit. No one wants accidental changes, and welding on the check code afterwards is surely harder than when first building the module - the developer knows all the code paths, inputs, and logic - they have to, to complete the module.